ipykernel/kernelbase.py

```diff
@@ -164,6 +168,25 @@ def dispatch_control(self, msg):
         sys.stderr.flush()
         self._publish_status(u'idle')
 
+    def should_handle(self, stream, msg, idents):
+        """Check whether a message should be handled
+        
+        Allows subclasses to prevent handling of certain messages (e.g. aborted requests).
```

Comment:
This only affects messages on the shell channel - it's not checked for the control channel. I think that's probably what we want, but maybe it should be mentioned in the docstring.